### ISSUE TITLE
memo does not follow dependencies

### ISSUE THREAD

**Issue by sant123** [fluid-js/fluid#2]
Hi, please see this code:

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Testing Fluid</title>
  </head>
  <body>
    <h1>Hello world</h1>
    <div id="app"></div>
    <script type="importmap">
      {
        "imports": {
          "fluid": "https://esm.sh/jsr/@fluid/core@1.0.8?dev"
        }
      }
    </script>
    <script type="module" src="main.js"></script>
  </body>
</html>

```

```js
import { memo, mount, state, tags } from "fluid";

const { button } = tags;

function Hello() {
  const counter = state(0);
  const counterPlusOne = memo(() => counter.val + 1);

  const handleClick = () => {
    counter.val++;
    console.log("Do click");
  };

  return button(
    {
      events: {
        click: handleClick,
      },
    },
    "Click me ",
    counter,
    " ",
    counterPlusOne,
  );
}

mount(document.getElementById("app"), Hello);

```

https://github.com/fluid-js/fluid/assets/7959437/21afcd4c-57e6-4e82-8a07-a94a12fd2f61

The expected behavior should be right?

- Click me 0 1
- Click me 1 2
- Click me 2 3
- Click me 3 4

Thanks!

**Comment by WorriedArrow** [comment]
It should be fixed in 1.0.9. Let me know if you have any further issues with this!

**Comment by sant123** [comment]
It works pretty good, @WorriedArrow thank you so much!!!
